sympy/plotting/plot.py

```diff
@@ -1068,15 +1073,33 @@ def __init__(self, parent):
                 self.ax[i].spines['right'].set_color('none')
                 self.ax[i].spines['bottom'].set_position('zero')
                 self.ax[i].spines['top'].set_color('none')
-                self.ax[i].spines['left'].set_smart_bounds(True)
-                self.ax[i].spines['bottom'].set_smart_bounds(False)
                 self.ax[i].xaxis.set_ticks_position('bottom')
                 self.ax[i].yaxis.set_ticks_position('left')
 
     def _process_series(self, series, ax, parent):
+        # ylim needs to be set in starting to get correct ybound and ylim
+        if parent.ylim:
+            from sympy.core.basic import Basic
+            ylim = parent.ylim
+            if any(isinstance(i,Basic) and not i.is_real for i in ylim):
+                raise ValueError(
+                "All numbers from ylim={} must be real".format(ylim))
+            if any(isinstance(i,Basic) and not i.is_finite for i in ylim):
+                raise ValueError(
+                "All numbers from ylim={} must be finite".format(ylim))
+            ylim = (float(i) for i in ylim)
+            ax.set_ylim(ylim)
```

Comment:
Why does this have to be set up first other?
And I see this is duplicate in line 1244 below.

Reply:
Some call in the implementation needed `ylim` to be set first because its value was to be used later on. It's been a while so I can't exactly recall which statement it was.